tests/test_array.h

```diff
@@ -0,0 +1,179 @@
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#ifndef TEST_ARRAY_H
+#define TEST_ARRAY_H
+
+#include "core/object/class_db.h"
+#include "core/object/script_language.h"
+#include "core/templates/hashfuncs.h"
+#include "core/templates/vector.h"
+#include "core/variant/array.h"
+#include "core/variant/container_type_validate.h"
+#include "core/variant/variant.h"
+#include "tests/test_macros.h"
+
+namespace TestArray {
+
+TEST_CASE("[Array] Size, Clear, and Empty") {
+	Array arr;
+	CHECK(arr.size() == 0);
+	CHECK(arr.empty());
+	arr.push_back(1);
+	CHECK(arr.size() == 1);
+	arr.clear();
+	CHECK(arr.empty());
+	CHECK(arr.size() == 0);
+}
+
+TEST_CASE("[Array] Assignment and Compare Operators") {
+	Array arr1;
+	Array arr2;
+	arr1.push_back(1);
+	CHECK(arr1 != arr2);
+	CHECK(arr1 > arr2);
+	CHECK(arr1 >= arr2);
+	arr2.push_back(2);
+	CHECK(arr1 != arr2);
+	CHECK(arr1 < arr2);
+	CHECK(arr1 <= arr2);
+	CHECK(arr2 > arr1);
+	CHECK(arr2 >= arr1);
+	Array arr3 = arr2;
+	CHECK(arr3 == arr2);
+}
+
+TEST_CASE("[Array] append_array") {
+	Array arr1;
+	Array arr2;
+	arr1.push_back(1);
+	arr1.append_array(arr2);
+	CHECK(arr1.size() == 1);
+	arr2.push_back(2);
+	arr1.append_array(arr2);
+	CHECK(arr1.size() == 2);
+	CHECK(int(arr1[0]) == 1);
+	CHECK(int(arr1[1]) == 2);
+}
+
+TEST_CASE("[Array] Resize, Insert, and Erase") {
+	Array arr;
+	arr.resize(2);
+	CHECK(arr.size() == 2);
+	arr.insert(0, 1);
+	CHECK(int(arr[0]) == 1);
+	arr.insert(0, 2);
+	CHECK(int(arr[0]) == 2);
+	arr.erase(2);
+	CHECK(int(arr[0]) == 1);
+}
+
+TEST_CASE("[Array] Front and Back") {
+	Array arr;
+	arr.push_back(1);
+	CHECK(int(arr.front()) == 1);
+	CHECK(int(arr.back()) == 1);
+	arr.push_back(3);
+	CHECK(int(arr.front()) == 1);
+	CHECK(int(arr.back()) == 3);
+}
+
+TEST_CASE("[Array] Has and Count") {
+	Array arr;
+	arr.push_back(1);
+	arr.push_back(1);
+	CHECK(arr.has(1));
+	CHECK(arr.has(2) == false);
```

Comment:
Not sure about this, but I usually prefer this way:

```suggestion
	CHECK(!arr.has(2));
```